gateway-service/src/main/java/org/java/mentorship/gateway/controller/NotificationController.java

```diff
@@ -0,0 +1,34 @@
+package org.java.mentorship.gateway.controller;
+
+
+import lombok.RequiredArgsConstructor;
+import org.java.mentorship.contracts.notification.client.NotificationFeignClient;
+import org.java.mentorship.contracts.notification.dto.Notification;
+import org.java.mentorship.contracts.notification.dto.NotificationChannel;
+import org.java.mentorship.contracts.notification.dto.NotificationType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/notifications")
+public class NotificationController {
+
+    private final NotificationFeignClient notificationFeignClient;
+
+    @GetMapping()
+    public ResponseEntity<List<Notification>> getNotifications(@RequestParam(required = false, name = "userId") Integer userId,
+                                                               @RequestParam(required = false, name = "email") String email,
+                                                               @RequestParam(required = false, name = "channel") NotificationChannel channel,
+                                                               @RequestParam(required = false, name = "type") NotificationType type,
+                                                               @RequestParam(required = false, name = "marked") Boolean marked) {
+        return ResponseEntity.ok(notificationFeignClient.getNotifications(userId, email, channel, type, marked));
+    }
+
+    @PostMapping("/{id}/read")
```

Comment:
This is not restful

I think this should be a PUT @ /{id} that can receive a whole Notification object but updates only the markedAsRead field